pkg/controller/ippool/common.go

```diff
@@ -25,7 +26,7 @@ func prepareAgentPod(
 	agentServiceAccountName string,
 	agentImage *config.Image,
 ) *corev1.Pod {
-	name := fmt.Sprintf("%s-%s-agent", ipPool.Namespace, ipPool.Name)
+	name := name.SafeConcatName(ipPool.Namespace, ipPool.Name)
```

Comment:
I have an IPPool in `harvester-public/vlan1`. For the initial design, the name is:
```
harvester-system    harvester-public-vlan1-agent
```

and the new implementation is
```
harvester-system    harvester-public-vlan1
```

Maybe we can still consider adding a prefix here to make the pod name make more sense (say it's a dhcp agent)

Reply:
I made a slight change to the SafeConcatName to incorporate the "-agent" suffix. PTAL, thanks!